gating/thaw/restart-nova-api-metadata.yml

```diff
@@ -0,0 +1,10 @@
+- name: Bring nova-api-metadata back online
+  hosts: nova_api_metadata
+  tasks:
+    - name: Wait for nova-api-metadata service to time out
+      pause:
+        minutes: 2
```

Comment:
seems racy...  Is there a reason why it has to be this way?

Reply:
So when we build a deploy from a saved image, the containers are all down and therefore have to be started.  When the nova_api_metadata container starts, the nova-api-metadata service times out after a little bit of time with this error:

```
2018-02-07 21:02:55.165 196 DEBUG oslo_messaging._drivers.amqpdriver [req-151ddc90-b579-447e-a4dd-8d8aac7623ef - - - - -] CALL msg_id: 2e9721b51aed495397917916d0e655bd exchange 'nova' topic 'conductor' _send /openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_messaging/_drivers/amqpdriver.py:568
2018-02-07 21:03:55.168 196 ERROR oslo_service.service [req-151ddc90-b579-447e-a4dd-8d8aac7623ef - - - - -] Error starting thread.
2018-02-07 21:03:55.168 196 ERROR oslo_service.service Traceback (most recent call last):
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_service/service.py", line 708, in run_service
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     service.start()
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/nova/service.py", line 357, in start
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     self.binary)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_versionedobjects/base.py", line 177, in wrapper
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     args, kwargs)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/nova/conductor/rpcapi.py", line 236, in object_class_action_versions
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     args=args, kwargs=kwargs)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_messaging/rpc/client.py", line 169, in call
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     retry=self.retry)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_messaging/transport.py", line 97, in _send
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     timeout=timeout, retry=retry)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 584, in send
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     retry=retry)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 573, in _send
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     result = self._waiter.wait(msg_id, timeout)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 456, in wait
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     message = self.waiters.get(msg_id, timeout=timeout)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service   File "/openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_messaging/_drivers/amqpdriver.py", line 347, in get
2018-02-07 21:03:55.168 196 ERROR oslo_service.service     'to message ID %s' % msg_id)
2018-02-07 21:03:55.168 196 ERROR oslo_service.service MessagingTimeout: Timed out waiting for a reply to message ID 2e9721b51aed495397917916d0e655bd
2018-02-07 21:03:55.168 196 ERROR oslo_service.service
2018-02-07 21:03:55.171 196 INFO nova.wsgi [-] Stopping WSGI server.
2018-02-07 21:03:55.171 196 DEBUG oslo_concurrency.lockutils [-] Acquired semaphore "singleton_lock" lock /openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_concurrency/lockutils.py:212
2018-02-07 21:03:55.172 196 DEBUG oslo_concurrency.lockutils [-] Releasing semaphore "singleton_lock" lock /openstack/venvs/nova-r14.7.0/lib/python2.7/site-packages/oslo_concurrency/lockutils.py:225
```
If you try to restart it manually before it has timed out, you basically just get that same error. What seems to work is to just give it some time to error itself, and then restart it again.  I know it's not the most elegant solution but I'm not entirely sure why it's erroring like that.  I've tried bringing the containers in different order (rabbit first, for example) but nothing seemed to help.

On a side note, when it times out, the process exits cleanly, so systemd doesn't try to restart it.

Reply:
sure, but is there no way other than a sleep?  what happens when we run this on a slower cloud?

Reply:
Not sure if this is why 2 mins makes, sense, but the timeout value for all the services are 2 mins: https://github.com/openstack/openstack-ansible-os_nova/blob/master/templates/nova-systemd-init.j2#L23

Reply:
ok, if the value were a var I'd say we should use that, but since it's hard coded this is fine

Reply:
If this startup issue is something not unique to this situation (imaging the host, etc) but instead a standard issue thing on any reboot, then it'd be good to get some sort of fix implemented upstream. I unfortunately can't think of anything off-hand though.